src/profile-logic/profile-data.js

```diff
@@ -797,6 +798,99 @@ export function getSearchFilteredMarkers(
+  const { stringTable, markers } = thread;
+  const newMarkers: MarkersTable = {
+    data: markers.data.slice(),
+    name: markers.name.slice(),
+    time: markers.time.slice(),
+    length: markers.length,
+  };
+
+  // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
+  // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
+  const bailoutRegex =
+    // Capture groups:
+    //       type   afterAt    where   bailoutLine  script functionLine
+    //        ↓     ↓          ↓             ↓        ↓    ↓
+    /^Bailout_(\w+) (after|at) ([\w ]+) on line (\d+) of (.*):(\d)+$/;
+
+  // Match: "Invalidate resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:3662"
+  // Match: "Invalidate self-hosted:4032"
+  const invalidateRegex =
+    // Capture groups:
+    //         url    line
+    //           ↓    ↓
+    /^Invalidate (.*):(\d)+$/;
+
+  const bailoutStringIndex = stringTable.indexForString('Bailout');
+  const invalidationStringIndex = stringTable.indexForString('Invalidate');
+  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+    const nameIndex = markers.name[markerIndex];
+    const time = markers.time[markerIndex];
+    const name = stringTable.getString(nameIndex);
+    let matchFound = false;
+    if (name.startsWith('Bailout_')) {
+      matchFound = true;
+      const match = name.match(bailoutRegex);
+      if (!match) {
+        console.error(`Could not match regex for bailout: "${name}"`);
+      } else {
+        const [
+          ,
+          type,
+          afterAt,
+          where,
+          bailoutLine,
+          script,
+          functionLine,
+        ] = match;
+        newMarkers.name[markerIndex] = bailoutStringIndex;
+        newMarkers.data[markerIndex] = ({
+          type: 'Bailout',
+          bailoutType: type,
+          afterAt: afterAt,
+          where: where,
+          script: script,
+          bailoutLine: +bailoutLine,
+          functionLine: +functionLine,
+          startTime: time,
+          endTime: time,
+        }: BailoutPayload);
+      }
+    } else if (name.startsWith('Invalidate ')) {
+      matchFound = true;
+      const match = name.match(invalidateRegex);
+      if (!match) {
+        console.error(`Could not match regex for bailout: "${name}"`);
+      } else {
+        const [, url, line] = match;
+        newMarkers.name[markerIndex] = invalidationStringIndex;
+        newMarkers.data[markerIndex] = {
+          type: 'Invalidation',
+          url,
+          line,
+          startTime: time,
+          endTime: time,
+        };
+      }
+    }
+    if (matchFound && markers.data[markerIndex]) {
+      console.error(
+        "A marker's payload was rewritten from its text, and it was assumed that " +
+          'the payload would be empty:',
```

Comment:
I don't find this error very clear. It's slightly too implicit that it's not empty.

Maybe this instead: 
```
Marker's payload was rewritten from its text, assuming that the payload would be empty,
but it isn't. Marker name is: <name>
```
(with whatever indentation you like, of course :) )